### PR TITLE
feat: added ability to share REPL using link

### PR DESCRIPTION
**Pull request by vankop**
> Which issue, if any, is this issue related to?

closes https://github.com/stylelint/stylelint-demo/issues/16

> Is there anything in the PR that needs further explanation?

I think nothing

### CODE REVIEW

**Comment by hudochenkov** [comment]
I tried this, and it works amazing!

What do you think if we change `/demo#?code=...&syntax=...&config=...` URL to simple encoded part `/demo#...`, where `...` encoded string? Similar to https://prettier.io/playground/ Here how they handle hash https://github.com/prettier/prettier/blob/master/website/playground/urlHash.js

**Comment by vankop** [comment]
Your idea is great, my approach was add additional flow for user:
- create REPL and share link **only** for config (css and syntax will be set to default)
  Where this approach could work -  `stylelint-config-standard`, `my-stylelint-config`, etc.
- craft link from 2 sources (with one hash user will open 2 tabs.. will need to merge data..)

In prepack case they have only JS there is no other params. A little bit ugly url I think does not make much sence

**Comment by hudochenkov** [comment]
I think this flows are very unlikely to happen. Our demo is not to share configs, but reproducible combination of config and styles.

https://eslint.org/demo also has only one string in URL.

**Comment by vankop** [comment]
> What do you think if we change `/demo#?code=...&syntax=...&config=...` URL to simple encoded part `/demo#...`

Done

File: src/common/utils.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import LZString from "lz-string";
+
+export function compress(data) {
+  try {
+    return LZString.compressToEncodedURIComponent(JSON.stringify(data));
+  } catch (e) {
```
Comment (hudochenkov) [inline src/common/utils.js]: Nitpick: Unused variable.

File: src/common/utils.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    return "";
+  }
+}
+
+export function decompress(str) {
+  try {
+    const data = JSON.parse(LZString.decompressFromEncodedURIComponent(str));
+
+    return typeof data !== "object" || data === null ? {} : data;
+  } catch (e) {
```
Comment (hudochenkov) [inline src/common/utils.js]: Nitpick: Unused variable.

File: src/common/utils.js (hunk 1 above)
Comment (vankop) [reply src/common/utils.js]: done

File: src/common/utils.js (hunk 2 above)
Comment (vankop) [reply src/common/utils.js]: done

**Pull request merged**